### PR TITLE
Remove double brackets from the ws url when using raw IPv6 address

### PR DESCRIPTION
**Pull request by ilkkao**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

TBD

### Motivation / Use-Case

Fixes: https://github.com/webpack/webpack-dev-server/issues/2943

### CODE REVIEW

File: client-src/default/utils/createSocketUrl.js [hunk 1]
```diff
@@ -72,14 +72,21 @@ function getSocketUrl(urlParts, loc) {
...
   if (portOption === 'location') {
     portOption = loc.port;
   }
 
+  // In case the host is a raw IPv6 address, it can be enclosed in
+  // the brackets as the brackets are needed in the final URL string.
+  // Need to remove those as url.format blindly adds its own set of brackets
+  // if the host string contains colons. That would lead to non-working
+  // double brackets (e.g. [[::]]) host
+  host = host.replace(/^\[(.*)\]$/, '$1');
```
Comment (alexander-akait) [inline client-src/default/utils/createSocketUrl.js:88]: Just interesting where we got extra brackets?

**Review by alexander-akait — APPROVED** [review]
Will be great to add tests, bug not high priority now

File: client-src/default/utils/createSocketUrl.js (hunk 1 above)
Comment (ilkkao) [reply client-src/default/utils/createSocketUrl.js:88]: In the problem case browser is navigated to `http://[::]:8080/` Websocket URL is formed by taking the current `location.hostname` which includes the brackets. Then in the end native `URL.format` method at least on Chrome V8 adds a new set of braces even they already exists.

Feels that this is a V8 issue and gets someday fixed there based on https://github.com/nodejs/node/issues/36654

**Comment by ilkkao** [comment]
Tested manually as I wasn't  able to quickly figure out how to add a new snapshot test. 

`http://[::]:8080/` url is used by the default on my Mac that has IPv6 enabled with the lastest webpack-dev-server beta. That makes this little more visible problem.

**Review by alexander-akait — APPROVED** [review]

**Comment by ilkkao** [comment]
Test failure btw doesn't seem to relate to my changes.

**Comment by thijstriemstra** [comment]
> http://[::]:8080/ url is used by the default on my Mac that has IPv6 enabled with the lastest webpack-dev-server beta. That makes this little more visible problem.

Running into same issue:

```
webpack 5.17.0 compiled successfully in 1183 ms
<i> [webpack-dev-server] Project is running at http://[::]:8080/
```

Will give this pr a try.

**Comment by alexander-akait** [comment]
Time to merge and move to new release

**Pull request merged**